src/main/obsRecorder.ts

```diff
@@ -174,44 +177,132 @@ const setOBSVideoResolution = (res: Size, paramString: string) => {
   const closestResolution = getClosestResolution(availableResolutions, res);
 
   setSetting('Video', paramString, closestResolution);
-}
+};
 
 /*
 * setupScene
 */
-const setupScene = (monitorIndex: number, baseResolution: Size, outputResolution: Size): IScene => {
+const setupScene = (options: RecorderOptionsType): IScene => {
+  const outputResolution = parseResolutionsString(options.obsOutputResolution);
+  let baseResolution: Size;
+
   setOBSVideoResolution(outputResolution, 'Output');
 
-  // Correct the monitorIndex. In config we start a 1 so it's easy for users. 
-  const monitorIndexFromZero = monitorIndex - 1;
-  console.info("[OBS] monitorIndexFromZero:", monitorIndexFromZero);
-  const selectedDisplay = displayInfo(monitorIndexFromZero);
-  if (!selectedDisplay) {
-    throw Error(`[OBS] No such display with index: ${monitorIndexFromZero}.`)
-  }
+  let videoSource: IInput;
 
-  setOBSVideoResolution(selectedDisplay.physicalSize, 'Base');
+  switch (options.obsCaptureMode) {
+    case 'monitor_capture':
+      // Correct the monitorIndex. In config we start a 1 so it's easy for users.
+      const monitorIndexFromZero = options.monitorIndex - 1;
+      console.info("[OBS] monitorIndexFromZero:", monitorIndexFromZero);
+      const selectedDisplay = displayInfo(monitorIndexFromZero);
+      if (!selectedDisplay) {
+        throw Error(`[OBS] No such display with index: ${monitorIndexFromZero}.`)
+      }
 
-  const videoSource = osn.InputFactory.create('monitor_capture', 'desktop-video');
+      baseResolution = selectedDisplay.physicalSize;
 
-  // Update source settings:
-  let settings = videoSource.settings;
-  settings['monitor'] = monitorIndexFromZero;
-  videoSource.update(settings);
-  videoSource.save();
+      videoSource = createMonitorCaptureSource(monitorIndexFromZero);
+      break;
+
+    case 'game_capture':
+      baseResolution = outputResolution;
+      videoSource = createGameCaptureSource();
+      break;
 
-  // A scene is necessary here to properly scale captured screen size to output video size
-  const scene = osn.SceneFactory.create('test-scene');
+    default:
+      throw Error(`[OBS] Invalid capture mode: ${options.obsCaptureMode}`);
+  }
+
+  setOBSVideoResolution(baseResolution, 'Base');
+
+  const scene: IScene = osn.SceneFactory.create('test-scene');
   const sceneItem = scene.add(videoSource);
   sceneItem.scale = { x: 1.0, y: 1.0 };
 
+  console.log(`[OBS] Configured video input source with mode '${options.obsCaptureMode}'`, inspectObject(videoSource.settings))
+
+  watchVideoSourceSize(sceneItem, videoSource, baseResolution);
```

Comment:
Do we need to call this for monitor capture? Seems like it might just be desired for game capture?

Reply:
Technically, we don't but if the resolution changes of the monitor even if the base resolution doesn't change, it will adjust the scaling so it still looks fine. I don't mind either way.